modules/stub/src/stub.js

```diff
@@ -1,241 +1,282 @@
 ((win, doc) => {
 
+  const TCF_API_NAME = '__tcfapi';
+  const POST_MESSAGE_EVENT = 'message';
+  const TCF_POST_MSG_CALL = TCF_API_NAME + 'Call';
+  const TCF_POST_MSG_RETURN = TCF_API_NAME + 'Return';
+  const TCF_LOCATOR_NAME = TCF_API_NAME + 'Locator';
+  const CMP_STATUS = 'stub';
+  const TARGET_ORIGIN = '*';
+  const PING_COMMAND = 'ping';
+  const GDPR_APPLIES_COMMAND = 'setGdprApplies';
+
   const makeStub = () => {
 
-    const TCF_API_NAME = '__tcfapi';
-    const EVENT_NAME = 'message';
-    const TCF_POST_MSG_CALL = TCF_API_NAME + 'Call';
-    const TCF_POST_MSG_RETURN = TCF_API_NAME + 'Return';
-    const TCF_LOCATOR_NAME = TCF_API_NAME + 'Locator';
-    const CMP_STATUS = 'stub';
-    const TARGET_ORIGIN = '*';
-    const PING_COMMAND = 'ping';
-    const GDPR_APPLIES_COMMAND = 'setGdprApplies';
+    const isLeader = (win === win.top && win[TCF_API_NAME] === undefined);
```

Comment:
I have a similar comment below... This assumes that the CMP would always be loaded on the page (top frame).  That doesn't seem like a safe assumption.  Instead, I think the __tcfapiLocator approach is needed to detect if a CMP exists in an ancestor frame.